### DISCUSSION TITLE
Output Signal

### DISCUSSION THREAD

**Discussion by dm4jh1** [EngineerGuy314/pico-WSPRer#39] [Ideas]
Hello, I think the WSPR tracker with only two components is brilliant. I would also like to fly the tracker. Now, the Raspberry Pi transmits square wave signals, with both antenna halves phase-shifted by 180 degrees. Have you thought about a simple low-pass filter made only of SMD parts? I know 30 mW is not much power, but I’m still hesitant without a filter. Can a classic filter be applied to both antenna outputs with your technology? I would be interested in your opinion.
73 de Jens DM4JH

**Comment by EngineerGuy314** [comment]
>>Can a classic filter be applied to both antenna outputs with your technology?

Absolutely it could. But I never bothered for two reasons. The output drivers on the RP2040 are not very good at RF frequencies, with a lot of slew-rate limitation.  So what actually comes out of them when under load is not really a square-wave.  The other reason is that the simple dipole antenna has a relatively narrow bandwidth, so a lot of the spurious RF energy never gets radiated. 

There are a few other low-powered trackers that use a SI5351 chip for output which also don't use filters. And a SI5351, which is actually intended to function at RF frequencies, puts out a much sharper square-wave then a microncontroller GPIO.

**Comment by dm4jh1** [comment]
Hello Robert,
I’ve now taken the time to take a closer look at the output signal.
First of all, you’re right, the curve does resemble a sine wave quite closely in the oscillogram.
That’s why the harmonics are well below 30 dB, so there shouldn’t be any issue.
However, more by coincidence, I examined the near field more closely with the spectrum analyzer.
There are significant sidebands (about 8 dB below the signal) – see spectrogram.
Do you have any idea where they could be coming from? They also appear next to the harmonics, very prominently. The only way to deal with them seems to be using a bandpass filter. I suspect they originate from the clock generation in the Pi.
As a filter, I was thinking of a resonant series circuit. What do you think?

73 de Jens (DM4JH)
![oszillogramm](https://github.com/user-attachments/assets/ba301570-96ae-47d5-a5c6-308ab1137881)
![spektrum](https://github.com/user-attachments/assets/f1ad877d-7d51-4506-b677-ed9477f0d520)

**Reply by serych** [reply]
Yes, you measured it exactly. The two peaks are in the signal because of the method used to generate the ultra-precise frequency. Try reading my description of the method, hopefully it will make sense to you: 
[https://github.com/serych/pico-WSPRer/wiki/how-does-pico-oscilator-work](https://github.com/serych/pico-WSPRer/wiki/how-does-pico-oscilator-work)  

Jakub

**Comment by EngineerGuy314** [comment]
DM4JH, what clock speed are you running your pico at? The higher the clock speed the cleaner the signal will be, and more energy will be focused in the fundamental frequency. 

Setting the clock (via 'K' in the setup menu) to 270Mhz would give the best performance, unfortunately not every pico can handle that level of overclocking. You need to try it, and if the pico locks up keep trying lower speeds. 180Mhz should be safe for all.

**Comment by dm4jh1** [comment]
Tnx Jakub and Rob,

oh yes I’m clocking the Pico at just 115 MHz to save some energy for the 300mA solar cells. I’ll try clocking the Pico higher and check the spectrum to see how many mA it consumes.

I will report...
